@here/harp-mapview/lib/MapView.ts

```diff
@@ -1454,6 +1467,16 @@ export class MapView extends THREE.EventDispatcher {
 
     /**
      * The THREE.js camera used by this `MapView` to render the main scene.
+     * @note When modifying the camera all derived properties like:
+     * - [[MapView.target]]
+     * - [[MapView.zoomLevel]]
+     * - [[MapView.tilt]]
+     * - [[MapView.heading]]
+     * could change.
+     * Some of the properties (like [[zoomLevel]]) are cached internaly
+     * and will only be updated in the next animation frame.
+     * FIXME: Unfortunatley THREE.js is not dispatching any events when camera properties change
+     * so we should have an API for enforcing update of cached values.
```

Comment:
I think that asks for api like `MapView.updateCamera` or `mapView.cameraChanged` that shall be called to sync internal cached params with camera when it was modified by user.

Having this, will also allow to cache `extractAttitude` result, so innocent people calls like `const { tilt, heading } = mapview` have expected complexity of just reading properties, not doing complex math twice.

Reply:
yes, this was my bigger plan. If you check https://github.com/heremaps/harp.gl/pull/1400 (sorry it is based on this PR and contains all these changes as well) or especially this commit:
https://github.com/heremaps/harp.gl/pull/1400/commits/ff2ae094af7aa6b3feaed6d6e72ae2bc754b0d82
I introduced a private `ViewState` in `MapView` and a method `MapView.updateViewState()`.

Things that I did not do yet are:
1. add tilt and heading to the `ViewState` so they are computed only once
2. Make `MapView.updateViewState` public